backend/src/test/java/capstone/backend/services/OrderToSupplierServiceTest.java

```diff
@@ -69,58 +74,93 @@ void createOrder() {
     void createOrderThrowsWhenProductNonExistent() {
         //GIVEN
         OrderToSupplierDTO orderToSave = sampleOrderDTO();
-        Long productId = orderToSave.getOrderItems().get(0).getProduct().getId();
-        when(productService.checkIfProductExists(productId)).thenReturn(false);
+        ProductDTO product = orderToSave.getOrderItems().get(0).getProduct();
+        when(productService.productExists(product)).thenReturn(false);
         //WHEN - //THEN
         Exception ex = assertThrows(IllegalArgumentException.class, () -> orderService.createOrder(orderToSave));
         assertThat(ex.getMessage(), is("You tried to order a product that doesn't exist!"));
-        verify(productService).checkIfProductExists(productId);
+        verify(productService).productExists(product);
     }
 
     @Test
     void createOrderThrowsWhenOrderAlreadyExists() {
         //GIVEN
         OrderToSupplierDTO orderToSave = sampleOrderDTO();
-        Long productId = orderToSave.getOrderItems().get(0).getProduct().getId();
-        when(productService.checkIfProductExists(productId)).thenReturn(true);
-        when(orderRepo.findById(orderToSave.getId())).thenReturn(Optional.of(mapOrder(orderToSave)));
+        when(orderRepo.existsById(orderToSave.getId())).thenReturn(true);
         //WHEN - //THEN
         Exception ex = assertThrows(EntityWithThisIdAlreadyExistException.class, () -> orderService.createOrder(orderToSave));
         assertThat(ex.getMessage(), is("An Order with this id already exists!"));
-        verify(productService).checkIfProductExists(productId);
-        verify(orderRepo).findById(orderToSave.getId());
+        verify(orderRepo).existsById(orderToSave.getId());
     }
+
     @Test
-    void createOrderThrowsWhenSupplierNonExitent(){
+    void createOrderThrowsWhenSupplierNonExitent() {
         //GIVEN
         OrderToSupplierDTO orderToSave = sampleOrderDTO();
-        Long productId = orderToSave.getOrderItems().get(0).getProduct().getId();
+        ProductDTO product = orderToSave.getOrderItems().get(0).getProduct();
         Long supplierId = orderToSave.getSupplier().getId();
-        when(productService.checkIfProductExists(productId)).thenReturn(true);
-        when(orderRepo.findById(orderToSave.getId())).thenReturn(Optional.empty());
-        when(supplierService.checkIfSupplierExists(supplierId)).thenReturn(false);
+        when(productService.productExists(product)).thenReturn(true);
+        when(orderRepo.existsById(orderToSave.getId())).thenReturn(false);
+        when(supplierService.supplierExists(supplierId)).thenReturn(false);
         //WHEN - //THEN
         Exception ex = assertThrows(IllegalArgumentException.class, () -> orderService.createOrder(orderToSave));
```

Comment:
I like to add a message for myself here to directly see in the console what went wrong:

```
Exception ex = assertThrows(IllegalArgumentException.class, () -> {
            orderService.createOrder(orderToSave);
        }, "I expected an IllegalArgumentException");
```

Reply:
you will see in the logs something like expected to throw ... anyways